src/Tribe/Plugin_Dependencies.php

```diff
@@ -0,0 +1,183 @@
+<?php defined( 'WPINC' ) or die;
+
+if (! class_exists( 'Tribe__Plugin_Dependencies' ) ) {
+	/**
+	 * Tracks which tribe plugins are currently activated
+	 */
+	class Tribe__Plugin_Dependencies {
+
+		/**
+		 * An multidimensional array of active tribe plugins in the following format
+		 *
+		 * array(
+		 *  'class'   => 'main class name',
+		 *  'version' => 'version num',
+		 *  'path'    => 'Path to the main plugin/bootstrap file'
+		 * )
+		 */
+		protected $active_plugins = array();
+
+		protected static $instance;
+
+		/**
+		 * Static Singleton Factory Method
+		 *
+		 * @return Tribe__Plugin_Dependencies
+		 */
+		public static function instance() {
+			if ( ! isset( self::$instance ) ) {
+				$className      = __CLASS__;
+				self::$instance = new $className;
+			}
+
+			return self::$instance;
+		}
+
+		public function __construct() {
+			$this->add_legacy_plugins();
+		}
+
+
+		/**
+		 * Registers older plugins that did not use this class
+		 */
+		private function add_legacy_plugins () {
+			// Version 4.2 and under of the plugins do not register themselves here, so we'll register them
+			// @TODO Consider removing this in 5.0
+
+			$tribe_plugins = new Tribe__Plugins_List();
+
+			foreach( $tribe_plugins->get_list() as $plugin ) {
+				if ( ! class_exists( $plugin[ 'class' ] ) ) continue;
+
+				$version = constant( $plugin[ 'class' ] . '::VERSION' );
+
+				$this->add_active_plugin( $plugin[ 'class' ], $version );
```

Comment:
Path is not required, but will be added by any extension in the future which uses the run plugin function created below. If everything goes according to plan all of our extensions, and eventually all of our plugins should be using it and thus will be adding their path. There are a number of scenarios where it could be useful, for things such as plugin_info() and deactivate_plugins() which are very probably things that we will eventually be using as part of our communication with tec servers and keeping extensions up to date.

I added some documentation at the top as well indicating it's optional. But I'd still like to collecting that info wherever possible.